hudi-utilities/src/main/java/org/apache/hudi/utilities/UtilHelpers.java

```diff
@@ -96,19 +94,21 @@
   private static final Logger LOG = LogManager.getLogger(UtilHelpers.class);
 
   public static Source createSource(String sourceClass, TypedProperties cfg, JavaSparkContext jssc,
-                                    SparkSession sparkSession, SchemaProvider schemaProvider, HoodieDeltaStreamerMetrics metrics) throws IOException {
-
+                                    SparkSession sparkSession, SchemaProvider schemaProvider,
+                                    HoodieDeltaStreamerMetrics metrics) throws IOException {
     try {
-      if (JsonKafkaSource.class.getName().equals(sourceClass)
-              || AvroKafkaSource.class.getName().equals(sourceClass)) {
+      try {
```

Comment:
I am not sure if handling this via exception is making this more readable. wdyt? 
For e.g the current impl makes it clear, in what case a certain constructor is called,.

Reply:
Not sure about readability, I would say it looks fine to me. The goal is to add an ability to override the parent class that expects "metrics" as a parameter, otherwise I simply can't do that and I can't include that class into the hudi source.

Reply:
Okay my bad. I understand what you tried to do now. So you have your own source class, that implements has a constructor with the metrics argument. 
I was thinking along the lines of having all sources take in the metrics field. We can deal with that separately